nntp/nntp.py

```diff
@@ -1741,83 +1741,85 @@ def post(
             for group in nntp_client.list("NEWSGROUPS"):
                 print(group)
         except NNTPError as e:
-            log("%s\n" % e)
+            log(f"{e}\n")
         log("\n")
 
         log("LIST NEWSGROUPS alt.binaries.*\n")
         try:
             groups = nntp_client.list("NEWSGROUPS", "alt.binaries.*")
             log("Entries %d\n" % len(list(groups)))
         except NNTPError as e:
-            log("%s\n" % e)
+            log(f"{e}\n")
         log("\n")
 
         log("LIST OVERVIEW.FMT\n")
         try:
-            log("%s\n" % list(nntp_client.list("OVERVIEW.FMT")))
+            log("{}\n".format(list(nntp_client.list("OVERVIEW.FMT"))))
         except NNTPError as e:
-            log("%s\n" % e)
+            log(f"{e}\n")
         log("\n")
 
         log("LIST HEADERS\n")
         try:
-            log("%s\n" % list(nntp_client.list("HEADERS")))
+            log("{}\n".format(list(nntp_client.list("HEADERS"))))
         except NNTPError as e:
-            log("%s\n" % e)
+            log(f"{e}\n")
         log("\n")
 
         log("LIST EXTENSIONS\n")
         try:
-            log("%s\n" % list(nntp_client.list("EXTENSIONS")))
+            log("{}\n".format(list(nntp_client.list("EXTENSIONS"))))
         except NNTPError as e:
-            log("%s\n" % e)
+            log(f"{e}\n")
         log("\n")
 
         log("POST (with illegal characters)\n")
         try:
             log(
-                "%s\n"
-                % nntp_client.post(
-                    HeaderDict(
-                        {
-                            "From": '"pynntp" <pynntp@not.a.real.doma.in>',
-                            "Newsgroups": "misc.test",
-                            "Subject": "pynntp test article",
-                            "Organization": "pynntp",
-                        }
-                    ),
-                    b"pip install pynntp\r\nthis\0contains\rillegal\ncharacters",
+                "{}\n".format(
+                    nntp_client.post(
+                        HeaderDict(
+                            {
+                                "From": '"pynntp" <pynntp@not.a.real.doma.in>',
+                                "Newsgroups": "misc.test",
+                                "Subject": "pynntp test article",
+                                "Organization": "pynntp",
+                            }
+                        ),
+                        b"pip install pynntp\r\nthis\0contains\rillegal\ncharacters",
+                    )
```

Comment:
These are a bit weird but I plan to remove the whole `if __name__ == "__main__"`:  block after most of this has been moved to proper integration tests